include/realtime_tools/realtime_server_goal_handle.h

```diff
@@ -135,36 +146,41 @@ class RealtimeServerGoalHandle
 
   void runNonRealtime()
   {
-    if (valid())
+    if (!valid())
+      return;
+
+    std::lock_guard<std::mutex> guard(mutex_);
+
+    try
     {
-      try
+      if (req_execute_ && !gh_->is_executing() && gh_->is_active() && !gh_->is_canceling())
```

Comment:
I know it was in the original change already, but I have a hard time to believe that I can trigger `execute` on a goal which is in either of the other states ('executing, !active, canceling`). 

Also, would it make sense to check if the atomics are set here to guarantee a realtime behavior?

Reply:
see https://github.com/ros-controls/realtime_tools/pull/57#issuecomment-603634364